source/common/router/retry_state_impl.cc

```diff
@@ -493,6 +495,10 @@ RetryStateImpl::wouldRetryFromReset(const Http::StreamResetReason reset_reason,
     return RetryDecision::RetryWithBackoff;
   }
 
+  if (retry_on_ & RetryPolicy::RETRY_ON_RESET_BEFORE_REQUEST) {
```

Comment:
I think we want to pass upstream response status here and only return this case if no response started